src/libraries/System.Net.Http.Json/tests/FunctionalTests/JsonContext/Person.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Net.Http.Json.Functional.Tests
+{
+    internal partial class JsonContext : JsonSerializerContext
+    {
+        private JsonTypeInfo<Person> _Person;
+        public JsonTypeInfo<Person> Person
+        {
+            get
+            {
+                if (_Person == null)
+                {
+                    JsonConverter customConverter;
+                    if (Options.Converters.Count > 0 && (customConverter = GetRuntimeProvidedCustomConverter(typeof(Person))) != null)
+                    {
+                        _Person = JsonMetadataServices.CreateValueInfo<Person>(Options, customConverter);
+                    }
+                    else
+                    {
+                        JsonTypeInfo<Person> objectInfo = JsonMetadataServices.CreateObjectInfo<Person>();
+                        _Person = objectInfo;
+
+                        JsonMetadataServices.InitializeObjectInfo(
+                            objectInfo,
+                            Options,
+                            createObjectFunc: static () => new Person(),
+                            PersonPropInitFunc,
+                            default);
+                    }
+                }
+
+                return _Person;
+            }
+        }
+        private static JsonPropertyInfo[] PersonPropInitFunc(JsonSerializerContext context)
+        {
+            JsonContext jsonContext = (JsonContext)context;
```

Comment:
Can this cast ever fail?

Reply:
It should never, unless we have a bug (e.g. a weird cross-pollination of options instances being used). The serializer should only pass the same context instance that the type info instance was a property on.

I'll think about this some more. We could be more defensive in the gen'd code and throw `InvalidOperationException` instead of leak `InvalidCastException`. 

To avoid any potential issues here entirely, we could make the method non-static and reference the context directly (instead of taking it as an argument), but I want to be sure of any perf implications first.